packages/twenty-server/src/engine/api/graphql/workspace-query-runner/services/record-position-backfill-service.ts

```diff
@@ -8,44 +8,92 @@ import {
   RecordPositionQueryType,
 } from 'src/engine/api/graphql/workspace-query-builder/factories/record-position-query.factory';
 import { RecordPositionFactory } from 'src/engine/api/graphql/workspace-query-runner/factories/record-position.factory';
+import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
 
 @Injectable()
 export class RecordPositionBackfillService {
+  private readonly logger = new Logger(RecordPositionBackfillService.name);
   constructor(
+    private readonly objectMetadataService: ObjectMetadataService,
     private readonly recordPositionFactory: RecordPositionFactory,
     private readonly recordPositionQueryFactory: RecordPositionQueryFactory,
     private readonly workspaceDataSourceService: WorkspaceDataSourceService,
   ) {}
 
-  async backfill(
-    workspaceId: string,
-    objectMetadata: { nameSingular: string; isCustom: boolean },
-    recordId: string,
-  ) {
-    const position = await this.recordPositionFactory.create(
-      'last',
-      objectMetadata as ObjectMetadataInterface,
-      workspaceId,
-    );
-
+  async backfill(workspaceId: string, dryRun: boolean) {
     const dataSourceSchema =
       this.workspaceDataSourceService.getSchemaName(workspaceId);
 
-    const [query, params] = await this.recordPositionQueryFactory.create(
-      {
-        recordPositionQueryType: RecordPositionQueryType.UPDATE_POSITION,
-        recordId,
-        positionValue: position,
-      },
-      objectMetadata as ObjectMetadataInterface,
-      dataSourceSchema,
-    );
-
-    this.workspaceDataSourceService.executeRawQuery(
-      query,
-      params,
-      workspaceId,
-      undefined,
-    );
+    const objectMetadataEntities =
+      await this.objectMetadataService.findManyWithinWorkspace(workspaceId, {
+        where: { isSystem: false },
+      });
+
+    const objectMetadataWithPosition =
+      objectMetadataEntities.filter(hasPositionField);
+
+    for (const objectMetadata of objectMetadataWithPosition) {
+      const [recordsWithoutPositionQuery, recordsWithoutPositionQueryParams] =
+        await this.recordPositionQueryFactory.create(
+          {
+            recordPositionQueryType: RecordPositionQueryType.FIND_BY_POSITION,
+            positionValue: null,
+          },
+          objectMetadata,
+          dataSourceSchema,
+        );
+
+      const recordsWithoutPosition =
```

Comment:
We lose typing here so we need to be careful for runtime errors, I'd check if recordsWithoutPosition is not empty / is an array maybe